testing-framework/src/lib.rs

```diff
@@ -0,0 +1,337 @@
+        (assigned_addr, pub_key_addr)
+    }
+
+    /// Deploy a new Actor at a given address, provided with a given token balance
+    /// and returns the CodeCID of the installed actor
+    fn deploy_actor_from_bin(
+        &mut self,
+        actor: &WasmActor,
+        balance: TokenAmount,
+    ) -> Result<Address, Error> {
+        let actor_id = rand::random();
+        let actor_address = Address::new_id(actor_id);
+
+        // Put the WASM code into the blockstore.
+        log::debug!("Deploying Actor {} code ", actor.name);
+        let code_cid = self
+            .state_tree
+            .store()
+            .put(
+                Code::Blake2b256,
+                &Block {
+                    codec: IPLD_RAW,
+                    data: &actor.code,
+                },
+            )
+            .setting_err(&actor.name)?;
+
+        // Add code cid to list of deployed contract
+        self.code_cids.push(code_cid);
+
+        // Set the Actor State on the `BlockStore`.
+        Self::set_actor(
+            &mut self.state_tree,
+            &actor.name,
+            &[(); 0],
+            code_cid,
+            actor_id,
+            0,
+            balance,
+        )?;
+
+        Ok(actor_address)
+    }
+
+    /// Deploy the main Actor file into the `StateTree`.
+    pub fn deploy_main_actor(&mut self, actor: WasmActor) -> Result<(), Error> {
+        let address = self.deploy_actor_from_bin(&actor, TokenAmount::zero())?;
+        self.main_actor_id = match address.id() {
+            Ok(id) => Some(id.to_ne_bytes().to_vec()),
+            Err(_) => panic!("Actor Id should be valid"),
+        };
+
+        Ok(())
+    }
+
+    /// Test an Actor on a `MemoryBlockstore`.
+    pub fn test(&mut self, actor: WasmActor, test: WasmActor) -> Result<(), Error> {
+        // TODO: Should we clone the `StateTree` before each test run,
+        // and make our `Tester` stateless?
+
+        let main_actor_id = self
+            .main_actor_id
+            .as_ref()
+            .cloned()
+            .ok_or(Error::MissingActor)?;
+
+        let test_address = self.deploy_actor_from_bin(&test, TokenAmount::zero())?;
+
+        let root = self
+            .state_tree
+            .flush()
+            .expect("Should be able to acquire the root CID by flushing");
+
+        let blockstore = self.state_tree.store().clone();
+
+        let mut executor = Self::new_executor(blockstore, root, self.builtin_actors.root);
+
+        let message = Message {
+            from: self.account.1,
+            to: test_address,
+            gas_limit: 1000000000,
+            method_num: 1,
+            params: main_actor_id.into(),
+            ..Message::default()
+        };
+
+        log::info!("testing test {} to Actor {}", test.name, actor.name);
+        executor
+            .execute_message(message, ApplyKind::Explicit, 100)
+            .tester_err(&format!("Could not test the Actor: {}", actor.name))
+            .map(|_| ())
```

Comment:
For each test actos we will be using against our target actor there will be multiple tests. 

Which means that when we execute the `test()` method we should have a loop somewhere that's calling for all available methods exposed by our test file. 

If what kept you from doing it is the lack of information of which are the methods we should at least have a TODO in the code to tackle it when the time is right.

Reply:
yup, in line with the above comment, updated to reflect the updated usecase